core/src/common/ipc_session.c

```diff
@@ -0,0 +1,241 @@
+{
+    channel->Sockfd = sockfd;
+    channel->FromAddr = *fromAddr;
+    channel->AddrLen = addrLen;
+}
+
+int IPCSession_AddRequestChannel(IPCSessionID sessionID, int sockfd, const struct sockaddr * fromAddr, int addrLen)
+{
+    int result = -1;
+    IPCSession * session = NULL;
+    if ((session = FindSessionByID(sessionID)) != NULL)
+    {
+        AddChannel(&session->RequestChannel, sockfd, fromAddr, addrLen);
+        result = 0;
+    }
+    else
+    {
+        Lwm2m_Error("No session with ID %d found\n", sessionID);
+        result = -1;
+    }
+    return result;
+}
+
+int IPCSession_GetRequestChannel(IPCSessionID sessionID, int * sockfd, const struct sockaddr ** fromAddr, int * addrLen)
+{
+    int result = -1;
+    IPCSession * session = NULL;
+    if ((session = FindSessionByID(sessionID)) != NULL)
+    {
+        if ((sockfd != NULL) && (fromAddr != NULL) && (addrLen != NULL))
+        {
+            *sockfd = session->RequestChannel.Sockfd;
+            *fromAddr = &session->RequestChannel.FromAddr;
+            *addrLen = session->RequestChannel.AddrLen;
+            result = 0;
+        }
+        else
+        {
+            Lwm2m_Error("NULL parameter - could not return notification channel information\n");
+            result = -1;
+        }
+    }
+    else
+    {
+        Lwm2m_Error("No session with ID %d found\n", sessionID);
+        result = -1;
+    }
+    return result;
+}
+
+int IPCSession_AddNotifyChannel(IPCSessionID sessionID, int sockfd, const struct sockaddr * fromAddr, int addrLen)
+{
+    int result = -1;
+    IPCSession * session = NULL;
+    if ((session = FindSessionByID(sessionID)) != NULL)
+    {
+        AddChannel(&session->NotifyChannel, sockfd, fromAddr, addrLen);
+        result = 0;
+    }
+    else
+    {
+        Lwm2m_Error("No session with ID %d found\n", sessionID);
+        result = -1;
+    }
+    return result;
+}
+
+
+int IPCSession_GetNotifyChannel(IPCSessionID sessionID, int * sockfd, const struct sockaddr ** fromAddr, int * addrLen)
+{
+    int result = -1;
+    IPCSession * session = NULL;
+    if ((session = FindSessionByID(sessionID)) != NULL)
+    {
+        if ((sockfd != NULL) && (fromAddr != NULL) && (addrLen != NULL))
+        {
+            *sockfd = session->NotifyChannel.Sockfd;
+            *fromAddr = &session->NotifyChannel.FromAddr;
+            *addrLen = session->NotifyChannel.AddrLen;
+            result = 0;
+        }
+        else
+        {
+            Lwm2m_Error("NULL parameter - could not return notification channel information\n");
+            result = -1;
+        }
+    }
+    else
+    {
+        Lwm2m_Error("No session with ID %d found\n", sessionID);
+        result = -1;
+    }
+    return result;
+}
+
+IPCSessionID IPCSession_AssignSessionID(void)
+{
+    static IPCSessionID sessionID = -1;
+    if (sessionID == -1)
+    {
+        sessionID = getpid() * 7487;
```

Comment:
Why not? :)  It's just a suitably large prime number. However I'll add a comment to explain why I chose it.